pkg/metrics/firewall.go

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: MIT
+
+package metrics
+
+import (
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/fwaas/firewalls"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/klog"
+)
+
+var (
+	firewallAdminStateUp *prometheus.GaugeVec
+	firewallStatus       *prometheus.GaugeVec
+
+	// according to the nl_constants from https://github.com/openstack/neutron-fwaas/blob/stable/ocata/neutron_fwaas/services/firewall/fwaas_plugin.py
+	// we will get the following firewall states
+	firewallStates = []string{"ACTIVE", "DOWN", "ERROR", "INACTIVE", "PENDING_CREATE", "PENDING_UPDATE", "PENDING_DELETE"}
+
+	firewallLabels = []string{"id", "name", "description", "policyID", "projectID"}
+)
+
+func registerFirewallMetrics() {
+	firewallAdminStateUp = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: generateName("firewall_admin_state_up"),
+			Help: "Firewall status",
+		},
+		firewallLabels,
+	)
+	firewallStatus = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: generateName("firewall_status"),
+			Help: "Firewall status",
+		},
+		append(firewallLabels, "status"),
+	)
+
+	prometheus.MustRegister(firewallAdminStateUp)
+	prometheus.MustRegister(firewallStatus)
+}
+
+// PublishFirewallMetrics makes the list request to the firewall api and
+// passes the result to a publish function.
+func PublishFirewallMetrics(client *gophercloud.ServiceClient, tenantID string) error {
```

Comment:
```suggestion
func PublishFWaaSV1Metrics(client *gophercloud.ServiceClient, tenantID string) error {
```

We may later want to implement FWaaSV2 metrics too